The build script sets LC_ALL to POSIX, but the Endless systemd package
then creates /etc/locale.conf (and the /etc/default/locale symlink) to
set LANG=en_US.utf8. Since the en_US.utf8 locale is not created in the
build root, we then get a bunch of warnings setlocale failing.

Rather than try to get the locale defined, force the use of the POSIX
locale by setting it in the configuration.

--- a/build
+++ b/build
@@ -1830,6 +1830,12 @@ for SPECFILE in "${SPECFILES[@]}" ; do
 	fi
     fi
 
+    # Set the locale to POSIX in /etc/locale.conf to ensure it's used
+    # through the build
+    if test -f $BUILD_ROOT/etc/locale.conf ; then
+        echo "LC_ALL=POSIX" > $BUILD_ROOT/etc/locale.conf
+    fi
+
     if test "$BUILDTYPE" = spec ; then
 	TOPDIR=`chroot $BUILD_ROOT su -c "rpm --eval '%_topdir'" - $BUILD_USER`
 	if test -z "$TOPDIR"; then
